Core.ApplicationServices/UIConfiguration/UIModuleCustomizationService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.Authorization;
+using Core.ApplicationServices.Model.UiCustomization;
+using Core.ApplicationServices.Organizations;
+using Core.DomainModel.Organization;
+using Core.DomainModel.UIConfiguration;
+using Core.DomainServices.Authorization;
+using Core.DomainServices.Generic;
+using Core.DomainServices.Repositories.UICustomization;
+using Infrastructure.Services.DataAccess;
+
+namespace Core.ApplicationServices.UIConfiguration
+{
+    public class UIModuleCustomizationService : IUIModuleCustomizationService
+    {
+        private readonly ITransactionManager _transactionManager;
+        private readonly IOrganizationalUserContext _userContext;
+        private readonly IOrganizationService _organizationService;
+        private readonly IEntityIdentityResolver _identityResolver;
+        private readonly IUIModuleCustomizationRepository _repository;
+
+        public UIModuleCustomizationService(ITransactionManager transactionManager,
+            IOrganizationalUserContext userContext,
+            IOrganizationService organizationService,
+            IEntityIdentityResolver identityResolver,
+            IUIModuleCustomizationRepository repository)
+        {
+            _transactionManager = transactionManager;
+            _userContext = userContext;
+            _organizationService = organizationService;
+            _identityResolver = identityResolver;
+            _repository = repository;
+        }
+
+
+        public Result<UIModuleCustomization, OperationError> GetModuleConfigurationForOrganization(int organizationId, string module)
+        {
+            return GetOrganizationById(organizationId)
+                .Bind(organization => organization.GetUiModuleCustomization(module)
+                    .Match(
+                        Result<UIModuleCustomization, OperationError>.Success,
+                        () => new OperationError($"module customization {module} not found on organization with id:{organizationId}", OperationFailure.NotFound)
+                    )
+                );
+        }
+
+        public Maybe<OperationError> UpdateModule(UIModuleCustomizationParameters parameters)
+        {
+            using var transaction = _transactionManager.Begin();
+            var error = GetOrganizationById(parameters.OrganizationId)
+                .Match(organization =>
+                    {
+                        if (!_userContext.HasRole(parameters.OrganizationId, OrganizationRole.LocalAdmin))
+                            return new OperationError("User is not a local admin in organization",
+                                OperationFailure.Forbidden);
+
+                        var nodesBefore = organization
+                            .GetUiModuleCustomization(parameters.Module)
```

Comment:
Add helper method in this class to get the nodes of a module.. then this block will not be repeated in line `70`